src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs

```diff
@@ -970,7 +970,64 @@ public static void Test(string name)
         {
             ""a"" => true,
             ""b"" => false,
-            _ => null,
+            _ => (bool?)null,
+        };
+    }
+}");
+        }
+
+        [WorkItem(37950, "https://github.com/dotnet/roslyn/issues/38771")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestShouldProperlyHandleInterfaces()
+        {
+            await TestInCSharp8(
+@"using System;
+
+class Program
+{
+    interface IFruit { }
+
+    interface IFruit2 { }
+
+    class Apple : IFruit, IFruit2 { }
+
+    class Banana : IFruit, IFruit2 { }
+
+    public static void Test(string name)
+    {
+        IFruit2 fruit;
+        [||]switch (name)
+        {
+            case ""apple"":
+                fruit = new Apple();
+            break;
+            case ""banana"":
+                fruit = new Banana();
+            break;
+            default:
+                throw new InvalidOperationException(""Unknown fruit."");
+        }
+    }
+}",
+@"using System;
+
+class Program
+{
+    interface IFruit { }
+
+    interface IFruit2 { }
+
+    class Apple : IFruit, IFruit2 { }
+
+    class Banana : IFruit, IFruit2 { }
+
+    public static void Test(string name)
+    {
+        var fruit = name switch
```

Comment:
Swich expressions are target-typed. Instead of casting arms, maybe it's better to just keep the type on the local declaration (`IFruit2 fruit = ...`) if needed.